packages/governance/src/types.js

```diff
@@ -32,3 +36,96 @@
  * @param {ParamDescriptions} paramDesc
  * @returns {ParamManagerFull}
  */
+
+/**
+ * @typedef {Object} BallotDetails
+ * @property {ChoiceMethod} method
+ * @property {string} question
+ * @property {string[]} positions
+ * @property {bigint} maxChoices
+ */
+
+/**
+ * @typedef {Object} Ballot
+ * @property {(positions: string[]) => CompletedBallot} choose
+ * @property {() => BallotDetails} getDetails
+ */
+
+/**
+ * @typedef {Object} PositionCount
+ * @property {string} position
+ * @property {number} tally
+ */
+
+/**
+ * @typedef {Object} VoteStatistics
+ * @property {number} spoiled
+ * @property {number} votes
+ * @property {PositionCount[]} results
+ */
+
+/**
+ * @typedef {Object} QurorumCounter
+ * @property {(VoteStatistics) => boolean} check
+ */
+
+/**
+ * @callback BuildBallot
+ * @param {ChoiceMethod} method
+ * @param {string} question
+ * @param {string[]} positions
+ * @param {bigint} maxChoices
```

Comment:
Do we want this to be a bigint, if we are comparing it to .length, a number? I think none of the code will error as is, but it might be nice for consistency